src/qmapshack/helpers/CMapIconSizesSetup.cpp

```diff
@@ -0,0 +1,83 @@
+/**********************************************************************************************
+    Copyright (C) 2014 Oliver Eichler <oliver.eichler@gmx.de>
+
+    This program is free software: you can redistribute it and/or modify
+    it under the terms of the GNU General Public License as published by
+    the Free Software Foundation, either version 3 of the License, or
+    (at your option) any later version.
+
+    This program is distributed in the hope that it will be useful,
+    but WITHOUT ANY WARRANTY; without even the implied warranty of
+    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+    GNU General Public License for more details.
+
+    You should have received a copy of the GNU General Public License
+    along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+**********************************************************************************************/
+
+#include "helpers/CMapIconSizesSetup.h"
+
+#include <QPushButton>
+#include <QSlider>
+
+#include "helpers/CSettings.h"
+#include "helpers/CWptIconManager.h"
+#include "poi/IPoiFile.h"
+
+CMapIconSizesSetup::CMapIconSizesSetup(QWidget* parent)
+    : QDialog(parent), imageWpt("://icons/48x48/Waypoint.png"), imagePoi("://icons/48x48/Poi.png") {
+  setupUi(this);
+
+  int labelSize = qMax(sliderWpt->maximum(), sliderPoi->maximum());
+  labelWpt->setFixedSize(labelSize, labelSize);
+  labelPoi->setFixedSize(labelSize, labelSize);
+
+  connect(sliderWpt, &QSlider::valueChanged, this, &CMapIconSizesSetup::slotSetIconSizeWpt);
+  connect(sliderPoi, &QSlider::valueChanged, this, &CMapIconSizesSetup::slotSetIconSizePoi);
+
+  SETTINGS;
+  sliderWpt->setValue(cfg.value("Icons/sizeWpt", 22).toInt());
```

Comment:
We have this 22 as magic value everywhere in the code now.  This is screaming for trouble if we want to change for 20 or 24. This should be a constexpr or a define. Best together with any other magic value in this patch.

The icon manager  header should be a good place.